internal/rabbitmq_client_factory.go

```diff
@@ -85,42 +71,63 @@ func generateRabbitholeClient(ctx context.Context, c client.Client, rmq topology
 		return nil, errors.New("failed to retrieve username: key password missing from secret")
 	}
 
-	rabbitmqClient, err := rabbithole.NewClient(endpoint, string(defaultUser), string(defaultUserPass))
+	if rmq.TLSEnabled() {
+		// create TLS config for https request
+		cfg := new(tls.Config)
+		cfg.RootCAs = certPool
+
+		transport := &http.Transport{TLSClientConfig: cfg}
+		rabbitmqClient, err = rabbithole.NewTLSClient(endpoint, string(defaultUser), string(defaultUserPass), transport)
+		if err != nil {
+			return nil, fmt.Errorf("failed to instantiate rabbit rabbitmqClient: %v", err)
+		}
+	} else {
+		rabbitmqClient, err = rabbithole.NewClient(endpoint, string(defaultUser), string(defaultUserPass))
+		if err != nil {
+			return nil, fmt.Errorf("failed to instantiate rabbit rabbitmqClient: %v", err)
+		}
+	}
+	return rabbitmqClient, nil
+}
+
+func managementEndpoint(cluster *rabbitmqv1beta1.RabbitmqCluster, svc *corev1.Service, hostname string) (string, error) {
+	port, err := managementPort(svc)
 	if err != nil {
-		return nil, fmt.Errorf("failed to instantiate rabbit rabbitmqClient: %v", err)
+		return "", err
 	}
 
-	return rabbitmqClient, nil
+	return fmt.Sprintf("%s://%s:%d", managementScheme(cluster), hostname, port), nil
+}
+
+// returns RabbitMQ management scheme from given cluster
+func managementScheme(cluster *rabbitmqv1beta1.RabbitmqCluster) string {
+	if cluster.TLSEnabled() {
+		return "https"
+	} else {
+		return "http"
+	}
 }
 
 // returns RabbitMQ management port from given service
 // if both "management-tls" and "management" ports are present, returns the "management-tls" port
 func managementPort(svc *corev1.Service) (int, error) {
+	var foundPort int
 	for _, port := range svc.Spec.Ports {
 		if port.Name == "management-tls" {
 			return int(port.Port), nil
 		}
 		if port.Name == "management" {
-			return int(port.Port), nil
+			foundPort = int(port.Port)
 		}
 	}
```

Comment:
what's the point of parameter `foundPort` here? You are also not using it in line 117. Could we get ride of `foundPort`?

Reply:
The order of `svc.Spec.Ports` is not guaranteed to be TLS first, so I was seeing cases where the management port would be found first and we'd return immediately, rather than finding the TLS port later.